gym/envs/robot_locomotion_group/drake/pusher_slider/pusher_slider_env.py

```diff
@@ -0,0 +1,253 @@
+
+        if "visualization" in config:
+            if not config["visualization"]:
+                pass
+            elif config["visualization"] == "meshcat":
+                self._sim_diagram.connect_to_meshcat()
+            elif config["visualization"] == "drake_viz":
+                self._sim_diagram.connect_to_drake_visualizer()
+            else:
+                raise ValueError("Unknown visualization:", config["visualization"])
+
+        self._sim_diagram.finalize()
+
+        builder = DiagramBuilder()
+        builder.AddSystem(self._sim_diagram)
+
+        pid = builder.AddSystem(PidController(kp=[0, 0], kd=[1000, 1000], ki=[0, 0]))
+        builder.Connect(self._sim_diagram.GetOutputPort("pusher_state_output"),
+                        pid.get_input_port_estimated_state())
+        builder.Connect(pid.get_output_port_control(),
+                        self._sim_diagram.GetInputPort("pusher_actuation_input"))
+        builder.ExportInput(pid.get_input_port_desired_state(),
+                            "pid_input_port_desired_state")
+
+        self._diagram = builder.Build()
+        self._pid_desired_state_port = self._diagram.get_input_port(0)
+        self._simulator = Simulator(self._diagram)
+        self.reset()
+
+        self.action_space = spaces.Box(low=-1., high= 1., shape=(2,),
+                                       dtype=np.float32)
+        # TODO: Observation space for images is currently too loose
+        self.observation_space = convert_observation_to_space(self.get_observation())
+
+    def get_observation(self, context=None):
+        assert self._sim_diagram.is_finalized()
+        if context is None:
+            context = self._simulator.get_context()
+        context = self._diagram.GetMutableSubsystemContext(self._sim_diagram, context)
+        mbp_context = self._sim_diagram.GetMutableSubsystemContext(
+            self._sim_diagram.mbp, context)
+        obs = dict()
+        obs["time"] = context.get_time()
+        obs["pusher"] = dict()
+        obs["pusher"]["position"] = np.copy(self._sim_diagram.mbp.GetPositions(
+            mbp_context, self._pusher_model_id))
+        obs["pusher"]["velocity"] = np.copy(self._sim_diagram.mbp.GetVelocities(
+            mbp_context, self._pusher_model_id))
+        q = np.copy(self._sim_diagram.mbp.GetPositions(mbp_context,
+                                                       self._slider_model_id))
+        v = np.copy(self._sim_diagram.mbp.GetVelocities(mbp_context,
+                                                        self._slider_model_id))
+        obs["slider"] = dict()
+        obs["slider"]["position"] = {'translation': q[4:], 'quaternion': q[0:4], 'raw': q}
+        obs["slider"]["velocity"] = {'linear': v[3:], 'angular': v[0:3], 'raw': v}
+        if self._config["rgbd_sensors"]["enabled"]:
+            obs["images"] = self._sim_diagram.get_image_observations(context)
+        return obs
```

Comment:
nice! I love the multiple observation options :+1:

Reply:
Thanks!